lib/config/index.js

```diff
@@ -96,11 +97,21 @@ config.isGitLabEnable = config.gitlab.clientID && config.gitlab.clientSecret
 config.isMattermostEnable = config.mattermost.clientID && config.mattermost.clientSecret
 config.isLDAPEnable = config.ldap.url
 config.isSAMLEnable = config.saml.idpSsoUrl
-config.isPDFExportEnable = config.allowpdfexport
+config.isPDFExportEnable = config.allowPDFExport
 
 // merge legacy values
-if (config.imageUploadType && !config.imageuploadtype) {
-  config.imageuploadtype = config.imageUploadType
+let keys = Object.keys(config)
+for (let i = keys.length; i--;) {
+  const uppercase = /[A-Z]/
+  // if the config contains uppercase letters
+  // and a lowercase version of this setting exists
+  // and the original, new, config with uppercase is not set
+  // in this case we set the new config using the old key.
+  if (uppercase.test(keys[i]) &&
+    config[keys[i].toLowerCase()] &&
+    !config[keys[1]]) {
+    config[keys[i]] = config[keys[i].toLowerCase()]
```

Comment:
This should also print a warning.

    `Warning: config.js contains deprecated lowercase setting for ${keys[i]}. Please change your config.js file to replace ${keys[i].toLowerCase()} with ${keys[i]}`

This way, we could remove the section at some point in the future. (think: warn for a year, break with warning for a year, remove completely)
Another possibility would be, to use this part to compare config.js' keys with a set of default keys - I see one upside and one downside to this:
- You now catch misspelling in the config
- You can't simply add a new config without also remembering that there's a default config file or config keys set.

Reply:
Yes, a warning should be added. :+1: 

I'll introduce a way to ignore invalid values and jump back to useful defaults in a future commit outside of this PR. for now I would like to transit to the camel case way of writing :)